### PR TITLE
Feature/support map types

### PR DESCRIPTION
**Pull request by dakrawczyk**
This pull request adds the ability for a map provider to declare what map types it supports, using `Provider.supportedMapTypes`.

The current options are: 
`MapType`
- `.normal`
- `.satellite`
- `.terrain`
- `.hybrid`

Based on that, Atlas will now allow the client to set what `MapType` it should display with `Atlas.currentMapType`. 

MapTypeSwitching

### CODE REVIEW

File: example/lib/main.dart [hunk 1]
```diff
@@ -132,4 +150,8 @@ class _AtlasSampleState extends State<AtlasSample> {
       ),
     );
   }
+
+  bool _shouldShowMapTypeButton() {
+    return (AtlasProvider.instance.supportedMapTypes.length > 1);
```
Comment (felangel) [inline example/lib/main.dart]:
```suggestion
    return AtlasProvider.instance.supportedMapTypes.length > 1;
```

File: packages/atlas/lib/src/atlas.dart [hunk 2]
```diff
@@ -60,13 +60,22 @@ class Atlas extends StatelessWidget {
   ///   * [showMyLocation] parameter.
   final bool showMyLocationButton;
 
+  /// Sets the underlying map type to be displayed.
+  ///
+  /// Defaults to [MapType.normal].
+  ///
+  /// See also:
+  ///   * [Provider.supportedMapTypes] parameter.
+  final MapType currentMapType;
```
Comment (felangel) [inline packages/atlas/lib/src/atlas.dart]:
```suggestion
  final MapType mapType;
```

File: packages/atlas/lib/src/provider.dart [hunk 3]
```diff
@@ -8,6 +8,14 @@ typedef void ArgumentCallback<T>(T argument);
 /// In order to implement a custom `AtlasProvider` you must simply implement `Provider`
 /// and set your `AtlasProvider.instance` to the instance of your custom `Provider`.
 abstract class Provider {
+  /// Allows the respective map provider to declare
+  /// what underlying map types are supported.
+  ///
+  /// This info can then be used to either cycle through
+  /// MapTypes with a button on the GUI, or to set the initial
+  /// [Atlas.currentMapType] property.
+  Set<MapType> supportedMapTypes;
```
Comment (felangel) [inline packages/atlas/lib/src/provider.dart]: why is this necessary? Shouldn't it be handled in the specific atlas packages?

File: example/lib/main.dart (hunk 1 above)
Comment (felangel) [reply example/lib/main.dart]: also if `supportedMapTypes` is just needed for this logic couldn't there be a `showMapTypeButton` flag similar to the `showLocationButton` flag?

**Review by felangel — CHANGES_REQUESTED** [review]
see comments

File: example/lib/main.dart (hunk 1 above)
Comment (dakrawczyk) [reply example/lib/main.dart]: In this example, that is all it's being used for. However, I added it for the flexibility of knowing what mapTypes the provider actually supports. 

This information could help to dynamically drive an interface like this: 
Screen Shot 2020-01-21 at 8 57 56 AM

It could also be used to drive a button to cycle through the available `mapTypes`, if a client wanted to have a button to loop through `.normal` > `.satellite` > `.terrain` > `.hybrid`. If you think it is helpful, I can create this experience in the example to demonstrate the capability.

**Comment by dakrawczyk** [comment]
Working to address coverage regression now.

**Review by noahmateen — DISMISSED** [review]

File: packages/google_atlas/lib/src/google_atlas.dart [hunk 4]
```diff
@@ -9,12 +9,22 @@ import 'package:google_maps_flutter/google_maps_flutter.dart' as GoogleMaps;
 
 /// `Atlas` Provider for Google Maps
 class GoogleAtlas extends Provider {
+  Set<MapType> get supportedMapTypes => {
```
Comment (felangel) [inline packages/google_atlas/lib/src/google_atlas.dart:13]:
```suggestion
  @override
  Set<MapType> get supportedMapTypes => {
```

**Review by felangel — DISMISSED** [review]

**Comment by dakrawczyk** [comment]
@noahmateen @kmjaris @felangel please have another look at this. 

My goal was to bump the version number as part of the mapTypes work which has already been reviewed. Unfortunately this meant bringing Google_Atlas up to speed with the latest and fixing the tests in Google_Atlas.

**Review by tchabot22 — APPROVED** [review]

**Review by noahmateen — APPROVED** [review]
